src/software/ai/hl/stp/play/ball_placement_play.cpp

```diff
@@ -38,50 +64,271 @@ void BallPlacementPlay::getNextTactics(TacticCoroutine::push_type &yield,
+                                                            Vector::createFromAngle(pass.passerOrientation()).normalize(
+                                                                    ROBOT_MAX_RADIUS_METERS * 2),
+                                                            pass.passerOrientation(), 0.0);
+                    tactics.emplace_back(move_passer_tactic);
+                }
+                if (receiver_robot.has_value()) {
+                    move_receiver_tactic->updateRobot(receiver_robot.value());
+                    move_receiver_tactic->updateControlParams(pass.receiverPoint(), pass.receiverOrientation(), 0.0);
+                    tactics.emplace_back(move_receiver_tactic);
+                }
+
+                LOG (DEBUG) << "stuck2";
+
+
+                CIRCLE_SHIT_YIELD(tactics);
+            } while (!move_receiver_tactic->done() || !move_passer_tactic->done());
+
+            auto receiver_tactic = std::make_shared<ReceiverTactic>(pass);
+            do {
+                if (robot.has_value() && receiver_robot.has_value()) {
+                    pass_ball_tactic->updateRobot(robot.value());
+                    pass_ball_tactic->updateControlParams(pass.passerPoint(), pass.receiverPoint(),
+                                                          pass.speed());
+                    receiver_tactic->updateRobot(receiver_robot.value());
+                    receiver_tactic->updateControlParams(pass, true);
+
+                    LOG (DEBUG) << "stuck passing";
+                    TacticVector tactics = {receiver_tactic, pass_ball_tactic};
+                    CIRCLE_SHIT_YIELD(tactics);
+                } else {
+                    break;
+                }
+            } while (!receiver_tactic->done());
+
+            do {
+                if (robot.has_value()) {
+                    place_ball_tactic->updateRobot(robot.value());
+                    place_ball_tactic->updateControlParams(pass.receiverPoint(),
+                                                           std::nullopt, true);
+
+                    CIRCLE_SHIT_YIELD({place_ball_tactic});
+                } else {
+                    if (world.gameState().getBallPlacementPoint().has_value()) {
+                        robot = world.friendlyTeam().getNearestRobot(world.gameState().getBallPlacementPoint().value());
+                    }
+                }
+            } while (!place_ball_tactic->done());
+
+            do {
+                if (receiver_robot.has_value()) {
+                    move_away_tactic->updateRobot(receiver_robot.value());
+                    move_away_tactic->updateControlParams(
+                            world.ball().position() -
+                            Vector::createFromAngle(receiver_robot->orientation()).normalize(
+                                    ROBOT_MAX_RADIUS_METERS * 2), receiver_robot->orientation(), 0.0);
+
+                    CIRCLE_SHIT_YIELD({move_away_tactic});
```

Comment:
nit: can this be more descriptive? something like `CIRCLE_ABSOLUTELY_SHIT_YIELD`?